controller/src/main/java/org/jboss/as/controller/extension/ExtensionRegistry.java

```diff
@@ -521,29 +526,21 @@ void addParsingNamespace(final String namespace) {
 
         @Override
         public Integer getManagementInterfaceMajorVersion() {
-            return majorVersion;
-        }
-
-        private void setMajorVersion(Integer majorVersion) {
-            this.majorVersion = majorVersion;
+            return version != null ? version.getMajor() : null;
         }
 
         @Override
         public Integer getManagementInterfaceMinorVersion() {
-            return minorVersion;
-        }
-
-        private void setMinorVersion(Integer minorVersion) {
-            this.minorVersion = minorVersion;
+            return version != null ? version.getMinor() : null;
```

Comment:
Not a big deal since this is public API, but we could replace these getters with a simple getVersion().
If not, remove the null check